roles/local-build/tasks/main.yml

```diff
@@ -1,92 +1,90 @@
+
+- name: Calculate total builds
+  set_fact:
+    total_builds: "{{ [git_url | length if git_url is iterable and git_url is not string else 1, git_branch | length if git_branch is iterable and git_branch is not string else 1] | max }}"
+
 - name: Build - Check compiled binary
   ansible.builtin.stat:
-    path: "{{ kaia_build_check_file_path }}"
-  register: kaia_build_check_file_stat
-
-- name: Build - Check skipping compile
-  ansible.builtin.set_fact:
-    kaia_build_required: True
-  when:
-    - kaia_build_check_file_stat.stat.exists == False or 
-      kaia_build_skip_if_exists == False
+    path: "{{ kaia_build_check_file_path }}-{{ item }}"
+  register: kaia_build_check_file_stats
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Clean the old directory
   ansible.builtin.file:
     state: absent
-    path: "{{ kaia_build_dir }}"
-  when:
-    - kaia_build_required
+    path: "{{ kaia_build_dir }}-{{ item }}"
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Git checkout
   ansible.builtin.git:
-    repo: "{{ kaia_build_remote_git_url }}"
-    dest: "{{ kaia_build_dir }}"
-    version: "{{ kaia_build_remote_git_branch }}"
-  when:
-    - kaia_build_required
+    repo: "{{ git_url[item % git_url | length] if git_url is iterable and git_url is not string else git_url }}"
+    dest: "{{ kaia_build_dir }}-{{ item }}"
+    version: "{{ git_branch[item % git_branch | length] if git_branch is iterable and git_branch is not string else git_branch }}"
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Compile kaia binary
   ansible.builtin.command: |
-    docker build --build-arg KLAYTN_STATIC_LINK=1 --build-arg DOCKER_BASE_IMAGE={{ kaia_build_docker_base_image }} -t {{ kaia_build_docker_builder_image }} --output output .
+    docker build --build-arg KLAYTN_STATIC_LINK=1 --build-arg DOCKER_BASE_IMAGE={{ kaia_build_docker_base_image }} -t {{ kaia_build_docker_builder_image }}-{{ item }} --output output .
   args:
-    chdir: "{{ kaia_build_dir }}"
+    chdir: "{{ kaia_build_dir }}-{{ item }}"
   environment:
     DOCKER_BUILDKIT: 1
-  when:
-    - kaia_build_required
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Compile homi binary locally
   ansible.builtin.command: |
     make homi
   args:
-    chdir: "{{ kaia_build_dir }}"
-  when:
-    - kaia_build_required
+    chdir: "{{ kaia_build_dir }}-{{ item }}"
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
```

Comment:
https://github.com/kaiachain/kaiaspray/pull/62#issuecomment-3209561164

> I thought about it but using different versions of homi would be a bit more complicated, because it will result in different genesis files.

I think so, it's a complicated. Then should we choose a homi bainary and use it when multiple mode? I don't have strong opinion, please choose according to your preference.
* Random
* Index zero
* Etc.